src/state_point.F90

```diff
@@ -214,18 +218,59 @@ subroutine write_state_point()
              group="tallies/tally" // to_str(i), length=t % n_nuclide_bins)
         deallocate(temp_array)
 
-        ! Write number of score bins, score bins, and moment order
+        ! Write number of score bins, score bins, user score bins
         call sp % write_data(t % n_score_bins, "n_score_bins", &
              group="tallies/tally" // to_str(i))
         call sp % write_data(t % score_bins, "score_bins", &
              group="tallies/tally" // to_str(i), length=t % n_score_bins)
-        call sp % write_data(t % moment_order, "moment_order", &
-             group="tallies/tally" // to_str(i), length=t % n_score_bins)
-
-        ! Write number of user score bins
         call sp % write_data(t % n_user_score_bins, "n_user_score_bins", &
              group="tallies/tally" // to_str(i))
 
+      ! Write explicit moment order strings for each score bin
+#ifdef HDF5
+        call sp % open_group("tallies/tally" // trim(to_str(i)) // "/moments")
+        call sp % close_group()
+#endif
+
+        k = 1
+        MOMENT_LOOP: do j = 1, t % n_user_score_bins
+          select case(t % score_bins(k))
+          case (SCORE_SCATTER_N, SCORE_NU_SCATTER_N)
+            moment_name = 'P' // to_str(t % moment_order(k))
+            call sp % write_data(moment_name, "order", &
+                 group="tallies/tally" // trim(to_str(i)) // &
+                 "/moments/" // trim(to_str(k)))
```

Comment:
It would look like
``` Fortran
call sp % write_data(moment_name, "order" // trim(to_str(k)), &
    group="tallies/tally" // trim(to_str(i)) // "/moments")
```

In my mind, they are equivalent, except using datasets only means you avoid creating N groups which saves a little bit of space and time. This is just a nitpick, so if you feel strongly we can leave as is. Eventually we should move to having an array of strings but we needn't do it as part of this PR.